Plutarch/DataRepr/Internal/Field.hs

```diff
@@ -103,89 +107,55 @@ instance
   ptoFields = ptoFields . pfromData
 
 {- |
-  Bind a HRec of named fields from a compatible type.
--}
-pletFields ::
-  forall a b as s.
-  ( PDataFields a
-  , as ~ (PFields a)
-  , BindFields as
-  ) =>
-  Term s a ->
-  (HRec (TermsOf s as) -> Term s b) ->
-  Term s b
-pletFields t =
-  runTermCont $
-    bindFields $ ptoFields t
-
-{- | Bind a HRec of the first N fields.
-
-  Always more efficient than binding all fields.
--}
-pletNFields ::
-  forall n a b s as.
-  ( PDataFields a
-  , as ~ (Take n (PFields a))
-  , BindFields as
-  ) =>
-  Term s a ->
-  (HRec (TermsOf s as) -> Term s b) ->
-  Term s b
-pletNFields t =
-  runTermCont $
-    bindFields $ to $ ptoFields t
-  where
-    to :: Term s (PDataRecord (PFields a)) -> Term s (PDataRecord as)
-    to = punsafeCoerce
+  Bind a HRec of named fields containing (at least) all the specified
+  fields.
+  This will bind the hull of the subset of fields provided,
 
-{- | Bind a HRec, dropping the first N fields.
+  for example, suppose that:
+  @PFields x ~ '["x" ':= x, "y" ':= y, "z" ':= z, "w" ':= w]@,
 
-  Usually more efficient than binding all fields.
+  then:
+  @pletFields @["y", "w"]@ will bind the fields @'["y", "z", "w"]@.
```

Comment:
This doesn't make sense. Why is `z` bound?

Reply:
currently the field binding doesn't do any skipping, you're right that we can do better

Reply:
I'll try and get this working

Reply:
BTW it's OK if the implementation doesn't make use of the fact that you only need some of the fields, since we can just fix that later in a non-breaking change.